src/model/mod.rs

```diff
@@ -42,6 +42,203 @@ fn print_boards(board1: &Vec<types::SubField>, board2: &Vec<types::SubField>) {
         "-------------------------------------------------------------");
 }
 
+/* Determines whether the chosen field is a valid one. Considers the
+   Moore neighborhood because the ships shouldn't collide. */
+fn valid_field(player: &types::Player, input: usize, ori: &String) -> bool {
```

Comment:
@JulianKniephoff ![ori](http://gameranx.com/wp-content/uploads/2016/06/OriandtheBlindForestDE.jpg)

On a serious note: why `&String` instead of `&str`?